test/README.md

```diff
@@ -13,45 +13,62 @@ with nested virtualization, refer to
 [this Fedora guide](https://docs.fedoraproject.org/en-US/quick-docs/using-nested-virtualization-in-kvm/index.html)
 for more details and recommendations on ensuring it is enabled correctly.
 
-## Introduction
+## Preparation and general invocation
 
-Before running the tests, ensure Cockpit has been built where the test suite
-expects to find it (do NOT run the build step as root):
+*Warning*: Never run the build, test, or any other command here as root!
 
-    $ ./test/image-prepare
+You first need to build cockpit, and install it into a VM:
 
-To run the integration tests run the following (do NOT run the integration tests
-as root):
+    $ test/image-prepare
 
-    $ ./test/common/run-tests
+This uses the default OS image, which is currently Fedora 35. See `$TEST_OS`
+below how to select a different one.
 
-The tests will automatically download the VM images they need, so expect
-that the initial run may take a couple of hours (there are quite a few
-images to retrieve for different scenario tests).
+In most cases you want to run an individual test:
+
+    $ test/verify/check-metrics TestCurrentMetrics.testCPU
+
+Or sometimes all tests in a single test file:
+
+    $ test/verify/check-session
+
+To see more verbose output from the test, use the `-v`/`--verbose` and/or `-t`/`--trace` flags:
+
+    $ test/verify/check-session --verbose --trace
+
+If you specify `-s`/`--sit` in addition, then the test will wait on failure and
+allow you to log into cockpit and/or the test instance and diagnose the issue.
+The cockpit and SSH addresses of the test instance will be printed:
 
-Alternatively you can run an individual test like this:
+    $ test/verify/check-session -st
 
-    $ ./test/image-prepare
-    $ ./test/verify/check-session
+You can also run *all* the tests, with some parallelism:
```

Comment:
well, you can also run one test with `test/common/run-tests` or any subset you add as arguments. But it is fine, normally users would not use this, so just saying it exists is fine.

Reply:
Right; I added that, but it's really not a tool that most developers should even know about.